### DISCUSSION TITLE
North American Month and Day

### DISCUSSION THREAD

**Discussion by Ken-Kauffmann** [arnaud-j4k08/PicoClockGreenEasy#1] [Q&A]
Currently, December 16th is displayed as "16 12". What sections of code would I modify in order for it to display as "12 16"?    I did not see anything obvious when scanning the source.  Thanks, Ken

**Comment by Ken-Kauffmann** [comment]
New to github, so I am unfamiliar with how to edit this.  It appears that the display of the date is done in Date.cpp, at stmts 49 and 56.  Day is drawn at horizontal position 0 in stmt 49, and month is drawn at horizontal position 13 in stmt 56.  Modifying 0 to 13 in stmt 49, and 13 to 0 in stmt 49 should reverse the displays.  This may well screw up the modification of day and month, so I will keep looking.  Thanks, Ken

**Reply by arnaud-j4k08** [reply]
Exactly! I was about to answer that you can simply swap the X-coordinates in these 2 lines. The same statements are called when modifying the date (that what the editedValueIndex and blinkingCounter parameters are used for), so it will work as well.

Actually, I realize that the dash separator suggests the ISO order YYYY-MM-DD, so I will change the default order to be MM-DD. If I find the time, I may also make the date format configurable.

Best regards,
Arnaud

**Reply by arnaud-j4k08** [reply]
I have pushed the corresponding changes to github. There is now a "Date format" entry in the "Options" menu. It allows choosing between 5 formats with different orders and separators.
